examples/platform/esp32/common/Esp32AppServer.cpp

```diff
@@ -158,17 +158,19 @@ void Esp32AppServer::Init(AppDelegate * sAppDelegate)
 {
     // Init ZCL Data Model and CHIP App Server
     static chip::CommonCaseDeviceServerInitParams initParams;
-#if CONFIG_TEST_EVENT_TRIGGER_ENABLED && CONFIG_ENABLE_OTA_REQUESTOR
+#if CONFIG_TEST_EVENT_TRIGGER_ENABLED
     if (hex_string_to_binary(CONFIG_TEST_EVENT_TRIGGER_ENABLE_KEY, sTestEventTriggerEnableKey,
                              sizeof(sTestEventTriggerEnableKey)) == 0)
     {
         ESP_LOGE(TAG, "Failed to convert the EnableKey string to octstr type value");
         memset(sTestEventTriggerEnableKey, 0, sizeof(sTestEventTriggerEnableKey));
     }
     static SimpleTestEventTriggerDelegate sTestEventTriggerDelegate{};
-    static OTATestEventTriggerHandler sOtaTestEventTriggerHandler{};
     VerifyOrDie(sTestEventTriggerDelegate.Init(ByteSpan(sTestEventTriggerEnableKey)) == CHIP_NO_ERROR);
```

Comment:
as a side comment, this isn't spec compliant.

Reply:
https://github.com/project-chip/connectedhomeip/blob/master/src/app/server/Server.cpp#L353 uses the `testEventTriggerDelegate` in the `SeviceInitParams`. So I have to enable test event trigger to use ICD feature. Otherwise the device will crash as `mTestEventTriggerDelegate` will be NULL.

Reply:
I was reffering to the key being in RAM and set this way. Spec requires the device to get the key from storage for certification as to not require the device to recomiled after certification since that is prohibited.

Havin the test event trigger enable for the ICD use-case is correct.

Reply:
OK ,I will raise another PR for it.